README.rst

```diff
@@ -1,11 +1,75 @@
+#########################
 frontend-component-header
-=========================
+#########################
 
 |Build Status| |Codecov| |npm_version| |npm_downloads| |license| |semantic-release|
 
-This is the standard Open edX header for use in React applications. It has two exports:
-  - **default**: The Header Component
-  - **messages**: for i18n in the form of ``{ locale: { key: translatedString } }``
+********
+Overview
+********
+
+A generic header for Open edX micro-frontend applications.
+
+************
+Requirements
+************
+
+This component uses ``@edx/frontend-platform`` services such as i18n, analytics, configuration, and the ``AppContext`` React component, and expects that it has been loaded into a micro-frontend that has been properly initialized via ``@edx/frontend-platform``'s ``initialize`` function. `Please visit the frontend template application to see an example. <https://github.com/edx/frontend-template-application/blob/master/src/index.jsx>`_
+
+Environment Variables
+=====================
+
+* ``LMS_BASE_URL`` - The URL of the LMS of your Open edX instance.
+* ``LOGOUT_URL`` - The URL of the API endpoint which performs a user logout.
+* ``LOGIN_URL`` - The URL of the login page where a user can sign into their account.
+* ``SITE_NAME`` - The user-facing name of the site, used as `alt` text on the logo in the header.
+  Defaults to "localhost" in development.
+* ``LOGO_URL`` - The URL of the site's logo.  This logo is displayed in the header.
+* ``AUTHN_MINIMAL_HEADER`` - A boolean flag which hides the main menu, user menu, and logged-out
+  menu items when truthy.  This is intended to be used in micro-frontends like
+  frontend-app-authentication in which these menus are considered distractions from the user's task.
+
+************
+Installation
+************
+
+To install this header into your Open edX micro-frontend, run the following command in your MFE:
+
+``npm i --save @edx/frontend-component-header``
+
+This will make the component available to be imported into your application.
+
+*****
+Usage
+*****
+
+This library has the following exports:
+
+* ``(default)``: The header as a React component.
+* ``messages``: Internationalization messages suitable for use with `@edx/frontend-platform/i18n <https://edx.github.io/frontend-platform/module-Internationalization.html>`_
+* ``dist/index.scss``: A SASS file which contains style information for the component.  It should be imported into the micro-frontend's own SCSS file.
+
+Examples
+========
+
+* `An example of component and messages usage. <https://github.com/edx/frontend-template-application/blob/master/src/index.jsx#L21>`_
+* `An example of SCSS file usage. <https://github.com/edx/frontend-template-application/blob/master/src/index.scss#L8>`_
```

Comment:
hmm, i wish these examples could be not links to files that may be changed. maybe don't link to master?

Reply:
Done!